setup.py

```diff
@@ -131,51 +163,84 @@ def load_description():
-            'falcon-bench = falcon.cmd.bench:main',
-            'falcon-inspect-app = falcon.cmd.inspect_app:main',
-            'falcon-print-routes = falcon.cmd.inspect_app:route_main',
-        ]
-    }
-)
+def run_setup(CYTHON):
+    if CYTHON:
+        cmdclass, ext_modules = get_cython_options()
+    else:
+        cmdclass, ext_modules = {}, []
+
+    setup(
+        name='falcon',
+        version=VERSION,
+        description='An unladen web framework for building APIs and app backends.',
+        long_description=load_description(),
+        long_description_content_type='text/x-rst',
+        classifiers=[
+            'Development Status :: 5 - Production/Stable',
+            'Environment :: Web Environment',
+            'Natural Language :: English',
+            'Intended Audience :: Developers',
+            'Intended Audience :: System Administrators',
+            'License :: OSI Approved :: Apache Software License',
+            'Operating System :: MacOS :: MacOS X',
+            'Operating System :: Microsoft :: Windows',
+            'Operating System :: POSIX',
+            'Topic :: Internet :: WWW/HTTP :: WSGI',
+            'Topic :: Software Development :: Libraries :: Application Frameworks',
+            'Programming Language :: Python',
+            'Programming Language :: Python :: Implementation :: CPython',
+            'Programming Language :: Python :: Implementation :: PyPy',
+            'Programming Language :: Python :: 3',
+            'Programming Language :: Python :: 3.5',
+            'Programming Language :: Python :: 3.6',
+            'Programming Language :: Python :: 3.7',
+            'Programming Language :: Python :: 3.8',
+        ],
+        keywords='wsgi web api framework rest http cloud',
+        author='Kurt Griffiths',
+        author_email='mail@kgriffs.com',
+        url='https://falconframework.org',
+        license='Apache 2.0',
+        packages=find_packages(exclude=['tests']),
+        include_package_data=True,
+        zip_safe=False,
+        python_requires='>=3.5',
+        install_requires=REQUIRES,
+        cmdclass=cmdclass,
+        ext_modules=ext_modules,
+        tests_require=['testtools', 'requests', 'pyyaml', 'pytest', 'pytest-runner'],
+        entry_points={
+            'console_scripts': [
+                'falcon-bench = falcon.cmd.bench:main',
+                'falcon-inspect-app = falcon.cmd.inspect_app:main',
+                'falcon-print-routes = falcon.cmd.inspect_app:route_main',
+            ]
+        }
+    )
+
+
+def status_msgs(*msgs):
+    print('*' * 75, *msgs, '*' * 75, sep='\n')
+
+
+if not CYTHON:
+    run_setup(False)
+    if not PYPY:
+        status_msgs('Cython compilation not supported in this environment')
+else:
+    try:
+        run_setup(True)
+    except BuildFailed as exc:
+        status_msgs(
+            exc.__cause__,
+            'Cython compilation could not be compiled, speedups are not enabled.',
```

Comment:
If we decide to continue with this, this sentence could probably be cleaned up, "compilation could not be compiled" (?)

Reply:
That's a typo. I think I meant "completed"